core/src/dht/dht_reducers.rs

```diff
@@ -230,57 +259,168 @@ pub mod tests {
+        let new_dht_store = reduce(store.dht(), &action_link_add);
+
+        let entry_link_remove = Entry::LinkRemove((
+            LinkData::from_link(
+                &link.clone(),
+                LinkActionKind::REMOVE,
+                test_chain_header(),
+                test_agent_id(),
+            ),
+            vec![entry_link_add.address()],
+        ));
+
+        //remove added link from dht
+        let action_link_remove = ActionWrapper::new(Action::RemoveLink(entry_link_remove.clone()));
+        let new_dht_store = reduce(new_dht_store, &action_link_remove);
 
-        let action = ActionWrapper::new(Action::RemoveLink(link.clone()));
-        let new_dht_store = reduce(new_dht_store, &action);
+        let link_type_regex = Regex::new("test-link").expect("Could not create link_type");
+        let tag_regex = Regex::new("test-tag").expect("Could not create link_type");
 
+        //fetch from dht and when tombstone is found return tombstone
         let storage = new_dht_store.meta_storage();
         let fetched = storage.read().unwrap().fetch_eavi(&EaviQuery::new(
             Some(entry.address()).into(),
-            EavFilter::predicate(|a| match a {
-                Attribute::LinkTag(_, _) | Attribute::RemovedLink(_, _) => true,
+            EavFilter::predicate(|attr: Attribute| match attr.clone() {
+                Attribute::LinkTag(query_link_type, query_tag)
+                | Attribute::RemovedLink(query_link_type, query_tag) => {
+                    link_type_regex.is_match(&query_link_type) && tag_regex.is_match(&query_tag)
+                }
                 _ => false,
             }),
             None.into(),
             IndexFilter::LatestByAttribute,
+            Some(EavFilter::single(Attribute::RemovedLink(
+                "test-link".to_string(),
+                test_tag.clone(),
+            ))),
         ));
 
+        //fetch call should be okay and remove_link tombstone should be the one that should be returned
         assert!(fetched.is_ok());
         let hash_set = fetched.unwrap();
         assert_eq!(hash_set.len(), 1);
         let eav = hash_set.iter().nth(0).unwrap();
         assert_eq!(eav.entity(), *link.base());
-        let link_entry = link.add_entry();
+        let link_entry = link.add_entry(test_chain_header(), test_agent_id());
         assert_eq!(eav.value(), link_entry.address());
         assert_eq!(
             eav.attribute(),
             Attribute::RemovedLink(link.link_type().to_string(), link.tag().to_string())
         );
+
+        //add new link with same chain header
+        let action_link_add = ActionWrapper::new(Action::AddLink(link_data));
+        let new_dht_store = reduce(store.dht(), &action_link_add);
+
+        //fetch from dht after link with same chain header is added
+        let storage = new_dht_store.meta_storage();
+        let fetched = storage.read().unwrap().fetch_eavi(&EaviQuery::new(
+            Some(entry.address()).into(),
```

Comment:
Please DRY up this big `EaviQuery` that gets constructed into a function, it gets used in several places and I'm afraid the test will go out of date if we change the implementation in `dht_store::get_links()`. OR, could you just use `get_links` in the test instead of constructing the `EaviQuery` manually here?